### DISCUSSION TITLE
Freezing layers YOLOv8

### DISCUSSION THREAD

**Discussion by eng-fabiorfo** [ultralytics/ultralytics#2513] [Docs]
Hello,

How can I freeze layers in yolov8, in case of transfer learning to train only a few layers on a custom dataset?

**Comment by chijiokeprince92** [comment]
I've been trying  to do the same but I found a reply that is helpful

**Reply by DavidZSchneider** [reply]
Could you share?

**Reply by binpengjie** [reply]
me too,could you share?

**Reply by pderrenger** [reply]
@binpengjie certainly! To freeze layers in YOLOv8 during transfer learning, you can use the `freeze` parameter in your training script. This parameter accepts either an integer or a list of integers. If you provide an integer, it will freeze the first N layers of the model. If you provide a list, it will freeze the layers at the indices specified in the list.

Here's an example of how to freeze layers:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8n.pt')  # Replace with your model of choice

# Freeze the first 10 layers
model.train(freeze=10)

# Alternatively, freeze specific layers by index
model.train(freeze=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
```

Remember to adjust the layer indices according to the architecture of the specific YOLOv8 model you are using. For more details on the training process and other parameters, you can refer to the [Train documentation](https://docs.ultralytics.com/modes/train). Happy training! 😊🚀
